pkg/visor/init.go

```diff
@@ -186,10 +187,39 @@ func initSNet(ctx context.Context, v *Visor, log *logging.Logger) error {
 		STCP: v.conf.STCP,
 	}
 
+	var nat stun.NATType
+	var host *stun.Host
+	var err error
+stunLoop:
+	for _, stunServer := range v.conf.StunServers {
+		nC := stun.NewClient()
+		nC.SetServerAddr(stunServer)
+
+		nat, host, err = nC.Discover()
+		if err != nil {
+			log.Warn(err)
+		}
+
+		switch nat {
+		case stun.NATError, stun.NATUnknown, stun.NATBlocked:
+			log.Warn(nat.String())
+			// incase we fail to connect to all the STUN servers give a warning
+			if stunServer == v.conf.StunServers[len(v.conf.StunServers)-1] {
+				log.Warn("All STUN servers offline")
```

Comment:
What will be the values of `nat` and `host` variables in this case? And what is expected behavior of sudph if we cannot determine nat type?

Reply:
`Nat` and `host` will have the value of whatever the result was with the STUN check. In that case, we don't need to show the host value anyway since we don't have any info on the client.

Reply:
We initilize the sudph if we are unable to determine the nat type.